registry/remote/repository.go

```diff
@@ -234,6 +236,28 @@ func (r *Repository) client() Client {
 	return r.Client
 }
 
+func (r *Repository) do(req *http.Request) (*http.Response, error) {
+	if r.HandleWarning == nil {
+		return r.client().Do(req)
+	}
+
+	resp, err := r.client().Do(req)
+	if err != nil {
+		return nil, err
+	}
+	warningHeaders := parseWarningHeaders(resp.Header[headerWarning])
+	for _, wh := range warningHeaders {
+		warning := Warning{
+			WarningHeader: wh,
+			Reference:     r.Reference,
+			RequestMethod: req.Method,
+			RequestPath:   req.URL.Path,
+		}
+		r.HandleWarning(warning)
+	}
```

Comment:
This part seems duplicated. Can it be refactored?

Reply:
Refactored.